src/detect-detection-filter.c

```diff
@@ -304,12 +305,11 @@ static int DetectDetectionFilterTestParse02 (void)
 {
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("track both,count 10,seconds 60");
-    if (df && (df->track == TRACK_DST || df->track == TRACK_SRC) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
+    FAIL_IF(df && (df->track == TRACK_DST || df->track == TRACK_SRC) && (df->count == 10) &&
```

Comment:
can you break these into 4 separate checks?

Reply:
I tried breaking it and when I run the test I get a segmentation fault and the flow breaks.

Reply:
`static int DetectDetectionFilterTestParse02 (void)
{
    DetectThresholdData *df = NULL;
    df = DetectDetectionFilterParse("track both,count 10,seconds 60");

    FAIL_IF_NOT_NULL(df);
    FAIL_IF(df->track == TRACK_DST || df->track == TRACK_SRC);
    FAIL_IF(df->count == 10);
    FAIL_IF(df->seconds == 60);

    DetectDetectionFilterFree(NULL, df);

    PASS;
}`